### ISSUE TITLE
Accessibility - FocusTrapCallout doesn't trap focus on iOS device when screen reader is on

### ISSUE THREAD

**Issue by yanbxice** [microsoft/fluentui#14127]
### Environment Information

- **Package version(s)**: version 7
- **Browser and OS versions**: Chrome and Safari, Voiceover, iOS 13.3

### Describe the issue:

FocusTrapCallout cannot trap the focus on iOS device. When I turn on the Voiceover on my iPhone and navigate on the screen by swiping left and right, I can always get out of the callout and focus on some other elements on the page.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/yanbxice/pen/VwermWg

#### Actual behavior:

Focus always get to other elements on the page after the last button on callout.

#### Expected behavior:

It should announce "no next item" and "no previous item" like on Windows, and never focus on others outside of the callout, when swipe right and left.

**Comment by paulgildea** [comment]
@smhigley - FYI

**Comment by xugao** [comment]
I think this is essentially the same issue as [this](https://github.com/microsoft/fluentui/issues/11241#issuecomment-558393305)? @smhigley can help confirm

**Comment by smhigley** [comment]
So I believe both of these issues come from incomplete support for `aria-modal="true"`, and it's worth nothing that support for `aria-modal` has been changing a lot recently, generally for the better.

I can't reproduce this issue with Safari + VO on iOS 13.5.1. The example is in an iframe, so I am able to exit the iframe to the other Codepen controls, but I can't access anything else within the same iframe when the callout is open. I also can't reproduce any issue with Narrator + Edge Chromium (v.86).

It does look like Scott O'Hara tested aria-modal with Safari/VO on iOS 13.3 (the same version in this issue), and found that it wasn't supported: https://scottaohara.github.io/testing/dialog-tests/. I'm guessing there's just been an update within the last couple minor versions to fix it :)

**Comment by xugao** [comment]
Closing this as external. @yanbxice - you can watch out for a fix in near-future iOS updates

**Comment by yanbxice** [comment]
I see. I noticed it is the same behavior after I upgraded to iOS 13.6, i.e. focus on another control outside of the iframe. Is it possible there can be a fix to even not focus on items out the iframe, jsut like what it does on Edge? Or this might be a Voiceover issue?
